include/swift/AST/Import.h

```diff
@@ -555,34 +558,35 @@ class DenseMapInfo<swift::ImportedModule> {
   }
 };
 
-template<>
-struct DenseMapInfo<swift::ImportedModuleDesc> {
-  using ImportedModuleDesc = swift::ImportedModuleDesc;
+template<typename ModuleInfo>
+struct DenseMapInfo<swift::AttributedImport<ModuleInfo>> {
+  using AttributedImport = swift::AttributedImport<ModuleInfo>;
 
-  using ImportedModuleDMI = DenseMapInfo<swift::ImportedModule>;
+  using ModuleInfoDMI = DenseMapInfo<ModuleInfo>;
   using ImportOptionsDMI = DenseMapInfo<swift::ImportOptions>;
   using StringRefDMI = DenseMapInfo<StringRef>;
+  // FIXME: SPI groups not used by DenseMapInfo???
```

Comment:
I was very surprised by this. @xymus, is this intentional?

Reply:
I'm not familiar with the DenseMapInfo logic and the doc is sparse so this was not intentional. However SPI groups on an import should affect type checking at the source file level only. Does it apply here?

Reply:
The only use of `DenseMapInfo` with `AttributedImport` keys that I can find is during cross-importing (in the form of a `SmallSetVector`). I think it could mean that, if you wrote something like:

```swift
@_spi(treasureMaps) import MapKit
@_spi(illuminatiBases) import MapKit
import SwiftUI
```

We would only consider the `@_spi(treasureMaps) import MapKit`, ignoring the second import.

Come to think of it, though, I don't think we've actually defined how cross-import overlays should interact with `@_spi`. I'll look into that on Monday—might ping you for input at that time.

Reply:
I'll include the SPI groups in `isEqual()`, but not in the hash value.